libs/rtemodel/include/RteKernel.h

```diff
@@ -127,18 +127,20 @@ class RteKernel
    * @param cprjFile project to load
    * @param toolchain toolchain to set
    * @param initialize flag to call InitializeCprj()
+   * @param generateRteHeaders whether to generate RTE headers or not
    * @return RteCprjProject pointer
   */
-  RteCprjProject* LoadCprj(const std::string& cprjFile, const std::string& toolchain = RteUtils::EMPTY_STRING, bool initialize = true);
+  RteCprjProject* LoadCprj(const std::string& cprjFile, const std::string& toolchain = RteUtils::EMPTY_STRING, bool initialize = true, bool generateRteHeaders = true);
 
   /**
    * @brief initializes loaded Rte
    * @param cprjProject pointer to RteCprjProject
    * @param toolchain compiler toolchain to select, if empty non-ambiguous default is used
    * @param toolChainVersion, use default if non-ambiguous
+   * @param generateRteHeaders whether to generate RTE headers or not
    * @return true if successful
   */
-  bool InitializeCprj(RteCprjProject* cprjProject, const std::string& toolchain = RteUtils::EMPTY_STRING, const std::string& toolChainVersion = RteUtils::EMPTY_STRING);
+  bool InitializeCprj(RteCprjProject* cprjProject, const std::string& toolchain = RteUtils::EMPTY_STRING, const std::string& toolChainVersion = RteUtils::EMPTY_STRING, bool generateRteHeaders = true);
```

Comment:
RteKernel is a singleton. There is no need to pass a boolean flag all way down in the hierarchy.
Instead RteKernel can contain methods like SetGenerateRteHeaders(bool)  and IsGenerateRteHeaders()  
or/and  IsCopyRteFiles(),  SetCopyRteFiles();

Reply:
> 3\. Restructuring InitFileInstance() and adding WriteInstanceFiles() : all cses should be considered, not only loading cprj file

Were there any specific cases you had in mind that wasn't covered?

Reply:
Unfortunately, other cases are outside this repository, but rely on current implementation :(
In particular, any call to RteProject::Update() should trigger file copy (if needed)
I do agree however that writing config files could be separated from updating internal structures.